src/components/Post.jsx

```diff
@@ -53,7 +58,35 @@ function Post(prop) {
     return 6;
   };
   const postHasImages = postImages.some((image) => image.length !== 0);
+ 
+  async function likeshandler() {
+    if (user && likecount !== undefined) {
+
+      const ind = likecount.indexOf(user.email);
+      if (ind !== -1) {
+        likecount.splice(ind, 1);
```

Comment:
I don't think it's a good idea to mutate prop values directly.

How about creating a copy of `likescount` prop first?
eg
```
...
const tmpLikesCount = [...likescount];

tmpLikesCount.push(...)
tmpLikesCount.splice(...)
...
```

Reply:
done.